app/app_xdg.go

```diff
@@ -21,6 +21,10 @@ import (
 	"fyne.io/fyne/v2/theme"
 )
 
+func init() {
+	internalapp.CurrentVariant.Store(uint64(findFreedesktopColorScheme()))
```

Comment:
This reverts the ability to load the app whilst DBus/Gnome calls are hanging.
Having pulled this out from the goroutine will return to non-gnome desktops having apps hang for 60 sec until the call returns.

No idea why, this just seems to be a thing that many Gnome apps do and it's not good that Fyne apps would do the same.

Reply:
Well, that change was also the one that broke theme lookup at startup but sure. I can do an ugly fix to repaint twice on startup instead.

Reply:
Of course ugly workarounds are not desirable.
If you have another way of avoiding the portal library calls hanging that would be better :)

Reply:
Ugly workaround might be a bit harsh. I have a fix ready now. Will look into the Dbus code to see if there is some flag to pass for it not to hang. Where is the hang happening? All non-Gnome, just FyneDesk or somethign else? If it only happens on desktops that do not support the appearance settings, I would guess it is a bug in my library :)

Reply:
> Ugly workaround might be a bit harsh.

Sorry, I thought I was re-using your words - I may have accidentally exaggerated them.

> All non-Gnome, just FyneDesk

I don't know about all, but XFCE is the same.

Reply:
> Sorry, I thought I was re-using your words - I may have accidentally exaggerated them.

I meant that my wordings for it was a bit too harsh. It is not as ugly of a fix as I thought :)

> I don't know about all, but XFCE is the same.

That sounds like it might be related to desktops not supporting the spec then. I have an idea incoming (no idea if it works or not) but lets merge this fix for now :)

Reply:
I would appreciate if you could try https://github.com/rymdport/portal/pull/12 together with a version of Fyne that doesn't have the cache. Might make life easier for us all :)

Reply:
> That sounds like it might be related to desktops not supporting the spec then. I have an idea incoming (no idea if it works or not) but lets merge this fix for now if it looks good :)

Possibly, but why would the method call hang instead of returning a message that it's not supported?

Reply:
Because I think it is trying to auto start the service but it fails to start (because it doesn't exist) so it has some kind of retry loop. That was my idea.